tests/SpecTests/Operation.php

```diff
@@ -43,6 +44,17 @@ private function __construct(stdClass $operation)
         }
     }
 
+    public static function fromCommandMonitoring(stdClass $operation)
+    {
+        $o = new self($operation);
+
+        if (isset($operation->collectionOptions)) {
+            $o->collectionOptions = (array) $operation->collectionOptions;
+        }
```

Comment:
This was originally missing. After fixing the extra keys logic in DocumentsMatchConstraint, the test in `unacknowledgedBulkWrite.json` began failing because the test runner never applied the write concern in `collectionOptions`. It's now handled properly.